README.md
# Aione Theme
This WordPress theme is designed to be a lightweight, secure theme based on Aione Framework

CHANGELOG.md
# CHANGELOG
Changelog for WordPress theme based on Aione Framework

## 7.0.1.6 06-09-2018
ADDED: Mobile Nav: Responsive menu for mobiles and tablets
FIXED: Blog Pagination Design

## 7.0.1.7 06-09-2018
UPDATED: Mobile Nav Design

## 7.0.1.8 06-09-2018
ADDED: Functionality Aione Data Table

## 7.0.1.9 06-09-2018
FIXED: Mobile nav z index was conflicting with slider

## 7.0.1.10 06-09-2018
FIXED: Menu alignment when menu is inside header

## 7.0.1.11 06-09-2018
FIXED: Mobile first sidebar layout

## 7.1.0.0 07-09-2018
RELEASE: 7.1.0.0 Stable Release

## 7.1.0.1 07-09-2018
UPDATED: Gravity Form CSS

## 7.1.0.2 07-09-2018
ADDED: Shortcode date
ADDED: Shortcode info

## 7.1.0.3 07-09-2018
ADDED: Powered by

## 7.1.0.4 08-09-2018
FIXED: Site title and tagline markups and css

## 7.1.0.5 13-09-2018
ADDED: Aione list
ADDED: CSS Styles
FIXED: Position Powered by

## 7.1.0.6 13-09-2018
UPDATED: Powered by link

## 7.1.0.7 19-09-2018
ADDED: Page padding option and css in global and per page options

## 7.1.0.8 22-09-2018
ADDED: Page content padding option and css in global and per page options

## 7.1.0.9 05-10-2018
FIXED: HTML Filters

## 7.1.0.10 05-10-2018
ADDED: Template Clean

## 7.1.0.11 22-10-2018
FIXED: Link on logo

## 7.1.0.12 23-10-2018
ADDED: Darlic theme styles for slider

## 7.1.0.13 10-11-2018
ADDED: User role(s) body classes

## 7.1.0.14 15-11-2018
ADDED: Slider capabilities

## 7.1.0.15 16-11-2018
ADDED: Post Nevigation links

## 7.1.0.16 30-11-2018
ADDED: Per page sidebar selection

## 7.1.0.17 03-12-2018
UPDATED: Custom Sidebars Design and functionality

## 7.1.0.18 03-12-2018
FIXED: Fixed Issue with custom sidebars

## 7.1.0.19 05-12-2018
ADDED: Social Share widget

## 7.1.0.20 10-12-2018
UPDATED: Footer and service worker

## 7.2.0.0 15-12-2018
ADDED: Added Aione Blocks for Gutenberg in 5.0

## 7.2.0.1 15-12-2018
FIXED: Fixed minor issue

## 7.2.0.2 21-12-2018
ADDED: Custom posts per page in archieve pagination
ADDED: Added post status in custom template
ADDED: added upcomming-tag shortcode
ADDED: Orderby and order fields in custom template
UPDATED: Blog shortcode

## 7.2.0.3 29-12-2018
FIXED: Fixed pagination

## 7.2.0.4 03-01-2019
ADDED: Custom template for media

## 7.3.0.0 12-01-2019
UPDATED: Revised all css and optimized for seo 

## 7.3.0.1 17-01-2019
UPDATED: Revised coding standards 

## 7.3.0.2 17-01-2019
UPDATED: Revised coding standards 

## 7.3.0.3 17-01-2019
UPDATED: Revised coding standards 

## 7.3.0.4 17-01-2019
FIXED: Fixed pagination issue

## 7.3.0.5 18-01-2019
UPDATED: Updated CSS 

## 7.3.0.6 24-01-2019
UPDATED: Updated Section Block

## 7.3.0.7 29-01-2019
ADDED: Aione List

## 7.3.0.8 30-01-2019
ADDED: Equal Heights

## 7.3.0.9 31-01-2019
ADDED: Flexbox CSS

## 7.3.0.10 01-02-2019
ADDED: Primary Nav Hover Animation

## 7.3.0.11 01-02-2019
FIXED: Blog Sidebar Issue
ADDED: Orange Theme to Aione tabs

## 7.3.0.12 04-02-2019
ADDED: Logo Alignment Option
ADDED: Export Reusable Blocks

## 7.3.0.13 06-02-2019
UPDATED: CSS Reorganization
UPDATED: Header Left/Right Design

## 7.3.0.14 06-02-2019
ADDED: Widget Font Size
ADDED: Widget Copyright Notice
ADDED: Widget Designed by

## 7.3.0.15 06-02-2019
FIXED: Enabled issue

## 7.3.0.16 15-02-2019
FIXED: Mobile Nav Issue
ADDED: Added ua-device-desktop/ua-device-mobile class to body by detecting the device

## 7.3.0.16 16-02-2019
UPDATED: Using Darlic CDN for loading CSS and JS

## 7.3.0.17 10-04-2019
UPDATED: Primary Menu Custom CSS

## 7.4.0.0 22-06-2019
UPDATED: Primary menu description CSS

## 7.4.0.1 02-07-2019
ADDED: Aione Star Rating

## 7.4.0.2 10-09-2019
ADDED: Aione Data Table Filtered Row Count

## 7.4.0.3 15-10-2019
ADDED: Aione Preloader
ADDED: Retina logo Support

## 7.4.0.4 04-11-2019
ADDED: Scroll Progress Bar

## 7.4.0.5 04-12-2019
ADDED: Google Analytics Code
FIXED: Per post setting js conflict

## 7.4.0.6 10-12-2019
ADDED: Responsive Slider

## 7.4.0.7 14-02-2020
ADDED: Underline Theme to Aione Tabs

## 7.5.0.0 08-04-2020
FIXED: Fixed for Block API in WordPress 5.4

## 7.5.0.1 06-10-2020
FIXED: Fixed form design

## 7.5.0.2 22-10-2020
UPDATED: Updated CSS classes

## 7.5.0.3 22-10-2020
ADDED: Archive Template Header and Footer

## 7.5.0.4 22-10-2020
ADDED: Shortcode to export as CSV 

## 7.6.0.0 29-01-2021
UPDATED: PHP 7.4 compatibility 

## 7.6.0.1 15-02-2021
UPDATED: Animate.css

## 7.7.0.0 06-06-2021
UPDATED: Custom CSS 

## 7.8.0.0 06-07-2021
ADDED: Structured data for archive page

## 7.8.0.1 06-08-2021
UPDATED: Aione Tabs Block 
UPDATED: Aione Column Block
REMOVED: Pattern
FIXED: Error in get_aione_page_settings 

## 7.8.0.2 26-08-2021
UPDATED: CSS Updates

## 7.8.0.3 03-09-2021
UPDATED: Aione Column Block recovery

## 7.8.0.4 11-09-2021
UPDATED: Aione Breadcrumbs widget : Home Icon

## 7.8.0.5 20-09-2021
UPDATED: Aione Button Block recovery

## 7.9.0.0 22-09-2021
ADDED: Lightbox

## 7.9.0.1 22-09-2021
FIXED: Block fixes

## 8.0.0.0 25-05-2022
UPDATED: Wordpress 6.0

functions.php
<?php
/**
* Aione functions and definitions
* @package Aione
*/


//Remove New Block Widget Support
add_filter( 'use_widgets_block_editor', '__return_false' );


function get_ajax_posts() {
	global $wp_query;
	if (isset($_REQUEST) ){
		$post_id = $_REQUEST['post_id'];
		$wp_query = new WP_Query($_REQUEST['query']); 
		$is_front_page = $_REQUEST['is_front_page'];
	} else {
		global $post; 
		global $wp_query;
		$post_id = $post->ID;
		$is_front_page = false;
	}

    
    $response = get_template_part('template/aione-content',null,array('post_id'=>$post_id,'wp_query'=>$wp_query,'is_front_page'=>$is_front_page ));

    echo($response);
    wp_die();
    //exit; // leave ajax call
}

// Fire AJAX action for both logged in and non-logged in users
add_action('wp_ajax_get_ajax_posts', 'get_ajax_posts');
add_action('wp_ajax_nopriv_get_ajax_posts', 'get_ajax_posts');

function get_ajax_filters() {
	if (isset($_REQUEST) ){
		$post_id = $_REQUEST['post_id'];
		$wp_query = new WP_Query($_REQUEST['query']); 
		$filters = $_REQUEST['filters'];
	}

	$response = get_template_part('template/aione-content',null,array('post_id'=>$post_id,'wp_query'=>$wp_query,'filters'=>$filters));

    echo($response);
    wp_die();
    // exit; // leave ajax call
}

// Fire AJAX action for both logged in and non-logged in users
add_action('wp_ajax_get_ajax_filters', 'get_ajax_filters');
add_action('wp_ajax_nopriv_get_ajax_filters', 'get_ajax_filters');


add_action( 'after_setup_theme', 'aione_init' );

if ( ! function_exists( 'aione_init' ) ) :

	function aione_init() {

		/*
		* textdomain for translation.
		*/
		load_theme_textdomain( 'aione', get_template_directory() . '/languages' );

		// This theme uses wp_nav_menu() in one location.
		register_nav_menus( array(
			'primary-menu' => esc_html__( 'Primary', 'aione' ),
		) );

		// Add default posts and comments RSS feed links to head.
		add_theme_support( 'automatic-feed-links' );

		// Page title tag in 
		add_theme_support( 'title-tag' );

		//Enable support for Post Thumbnails on posts and pages.
		add_theme_support( 'post-thumbnails' );

		// Add theme support for selective refresh for widgets.
		add_theme_support( 'customize-selective-refresh-widgets' );

		// Add support for full and wide align images.
		add_theme_support( 'align-wide' );

		// Add support for post formats.
		add_theme_support( 'post-formats', array( 'aside', 'image', 'video', 'quote', 'link', 'status' ) );

		// add block stypes
		// add_theme_support( 'wp-block-styles' );

		// Add support for custom color scheme.
		/*
		Depricated in 5.4
		See updated docs to manage
		add_theme_support( 'editor-color-palette',
			'#168dc5',
			'#1570a6',
			'#0073aa',
			'#229fd8',
			'#eee',
			'#444'
		);
		*/

		//Switch default core markup for search form, comment form, and comments to output valid HTML5
		add_theme_support( 'html5', array(
			'search-form',
			'comment-form',
			'comment-list',
			'gallery',
			'caption',
			'script',
			'style',
		) );

		// Set up the WordPress core custom background feature.
		/*
		add_theme_support( 'custom-background', apply_filters( '_s_custom_background_args', array(
			'default-color' => 'ffffff',
			'default-image' => '',
		) ) );
		*/

		/**
		 * Add support for core custom logo.
		 */
		/*
		add_theme_support( 'custom-logo', array(
			'height'      => 250,
			'width'       => 250,
			'flex-width'  => true,
			'flex-height' => true,
		) );
		*/

		remove_theme_support('core-block-patterns');

	}
endif;


//remove emoji styles and scripts
remove_action( 'wp_head', 'print_emoji_detection_script', 7 ); 
remove_action( 'admin_print_scripts', 'print_emoji_detection_script' ); 
remove_action( 'wp_print_styles', 'print_emoji_styles' ); 
remove_action( 'admin_print_styles', 'print_emoji_styles' );



/**
* Show Reusable blocks in wordpress export content area to export
*
* from one site to another site
*
*/
function aione_support_reusable_block_export( $post_type_args, $post_type ) {
	if ( 'wp_block' === $post_type ) {
		// $post_type_args['public'] = true;
		$post_type_args['can_export'] = true;
		// $post_type_args['show_in_menu'] = true;

		/*
		echo "<pre style='padding:50px 0 0 200px'>";
		print_r($post_type_args);
		echo "</pre>";
		*/

	}

	return $post_type_args;
}
add_filter( 'register_post_type_args', 'aione_support_reusable_block_export', 10, 2 );

/**
 * Set the content width in pixels, based on the theme's design and stylesheet.
 *
 * Priority 0 to make it available to lower priority callbacks.
 *
 * @global int $content_width
 */
function aionetheme_content_width() {
	$GLOBALS['content_width'] = apply_filters( 'aionetheme_content_width', 980 ); 
}
add_action( 'after_setup_theme', 'aionetheme_content_width', 0 );

add_filter( 'gform_init_scripts_footer', '__return_true' );
/*
function aione_remove_gf_scripts() {
    // De-register the default version of jQuery
    wp_deregister_script( 'jquery' );
    // Register a new version of jQuery under the same handle
    wp_register_script( 'jquery', 'https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js', false, '1.9.1', true );
}
add_action('wp_enqueue_scripts', 'aione_remove_gf_scripts', 10);
*/



/**
 * Enqueue scripts and styles.
 */
function aione_styles() {

	global $theme_options;

	$in_footer = false;
	if( $theme_options['advanced_load_js_bottom'] ){
		$in_footer = true;
	}


	wp_register_style( 'core', get_template_directory_uri() . '/assets/css/core.min.css', array(), null, 'all' );
	wp_register_style( 'aione', get_template_directory_uri() . '/assets/css/aione.min.css', array(), null, 'all' );
	wp_register_style( 'aionefull', get_template_directory_uri() . '/assets/css/aionefull.min.css', array(), null, 'all' );

	wp_register_script( 'aione-js', get_template_directory_uri() . '/assets/js/aione.min.js', array(), null, $in_footer );


	//If not loaded from CDN
	if( !$theme_options['advanced_use_cdn'] ) {

		if( $theme_options['advanced_load_css_bottom'] ) {
			//Enqueue core css essencial for layout in head, rest css will be loaded in footer
			wp_enqueue_style( 'core' );
		} else{
			wp_enqueue_style( 'aionefull' );
		}

		wp_enqueue_script( 'aione-js' );


	}
	

	// wp_deregister_style( 'wp-block-library' );
	wp_dequeue_style( 'wp-block-library' );
}
add_action( 'wp_enqueue_scripts', 'aione_styles' );
//add_action( 'admin_enqueue_scripts', 'aione_scripts' );




function aione_scripts() {

	global $theme_options;
	global $post;
	global $wp_query;

	//If not loaded from CDN
	if( !$theme_options['advanced_use_cdn'] ) {

		if( $theme_options['advanced_load_css_bottom'] ) {
			wp_enqueue_style( 'aione' );
		}
	}

	//wp_deregister_script( 'wp-embed' );
	wp_dequeue_script( 'wp-embed' );

	if ( is_singular() && comments_open() && get_option( 'thread_comments' ) ) {
		wp_enqueue_script( 'comment-reply' );
	} 
	if( $theme_options['advanced_ajax_content'] ) {
		
	?>
	<script>
		var post_id = <?php echo $post->ID; ?>;
		var query = <?php echo json_encode($wp_query ); ?>;
		var is_front_page = <?php echo json_encode($wp_query->is_front_page() ); ?>;
		console.log('query')
		console.log(query)
		console.log('is_front_page')
		console.log(is_front_page)
		//console.log(JSON.stringify(query))
		$.ajax({
	        type: 'POST',
	        url: '<?php echo admin_url('admin-ajax.php');?>',
	        //dataType: "html", // add data type
	        data: { action : 'get_ajax_posts', post_id : post_id, query:query.query, is_front_page:is_front_page },
	        success: function( response ) {
	        	//console.log('response')
	        	//console.log(response)
	        	//console.log(JSON.parse(response))
	        	//response = JSON.parse(response);
	            // $( '.aione-page-content > .wrapper' ).html( response ); 
	            $( '.content' ).html( response ); 
	        }
	    });
	</script>
	<?php
	}
}
add_action( 'get_footer', 'aione_scripts' );

/**
* Register styles for admin editor conponants
*/
add_action( 'admin_enqueue_scripts', 'aione_admin_editor_style' );
function aione_admin_editor_style( $hook ) {

	global $post_type;

	wp_register_style( 'aione-admin-css', get_template_directory_uri() . '/assets/css/admin/aione-admin.min.css', array(), false, 'all' );
    wp_enqueue_style( 'aione-admin-css' );


    if( in_array( $hook, array( 'post-new.php', 'post.php', 'widgets.php' ) ) ) {

		// If not the desired post type bail here.
		if( in_array( $post_type , array( 'aione-slider' ) ) ) {
			return;
		}

    	wp_register_style( 'aione-admin-editor-css', get_template_directory_uri() . '/assets/css/admin/aione-admin-editor.min.css', array(), false, 'all' );
    	wp_enqueue_style( 'aione-admin-editor-css' );

    }
    
}

/**
 * Register Google Fonts
 */
function aionetheme_fonts_url() {
	$fonts_url = '';

	/*
	 *Translators: If there are characters in your language that are not
	 * supported by Karla, translate this to 'off'. Do not translate
	 * into your own language.
	 */
	$notoserif = esc_html_x( 'on', 'Noto Serif font: on or off', 'aione' );

	if ( 'off' !== $notoserif ) {
		$font_families = array();
		$font_families[] = 'Noto Serif:400,400italic,700,700italic';
		$font_families[] = 'Open Sans:400,400italic,700,700italic';

		$query_args = array(
			'family' => urlencode( implode( '|', $font_families ) ),
			'subset' => urlencode( 'latin,latin-ext' ),
		);

		$fonts_url = add_query_arg( $query_args, 'https://fonts.googleapis.com/css' );
	}

	return $fonts_url;

}

/* Remove type='text/javascript' and type='text/css' for accesiblity */
//add_filter( 'style_loader_tag',  'clean_style_tag', 10, 2 );
//add_filter( 'script_loader_tag', 'clean_script_tag', 10, 2 );


/**
 * Clean up output of stylesheet <link> tags
 */

/*
function clean_style_tag( $input ) {
    preg_match_all( "!<link rel='stylesheet'\s?(id='[^']+')?\s+href='(.*)' type='text/css' media='(.*)' />!", $input, $matches );
    if ( empty( $matches[2] ) ) {
        return $input;
    }
    // Only display media if it is meaningful
    $media = $matches[3][0] !== '' && $matches[3][0] !== 'all' ? ' media="' . $matches[3][0] . '"' : '';

    return '<link rel="stylesheet" href="' . $matches[2][0] . '"' . $media . '>' . "\n";
}
*/
/**
 * Clean up output of <script> tags
 /*
function clean_script_tag( $input ) {
    $input = str_replace( "type='text/javascript' ", '', $input );

    return str_replace( "'", '"', $input );
}
*/

/**
 * Implement the Custom Header feature.
 * Removed on 2019/01/07
 * Delete this file if no issue
 */

// require get_template_directory() . '/includes/custom-header.php';

// Common Aione functions
require get_template_directory() . '/includes/aione-functions.php';

// Common Aione Theme functions
require get_template_directory() . '/includes/theme-functions.php';

// Common Aione Theme filters
require get_template_directory() . '/includes/theme-filters.php';

// Aione Widgets
require get_template_directory() . '/includes/aione-widgets.php';

// Register Sidebars(Widget Areas)
require get_template_directory() . '/includes/register-sidebars.php';

// Register Blocks(Using ACF)
require get_template_directory() . '/includes/aione-blocks.php';

// Register Block Patterns
//require get_template_directory() . '/includes/aione-block-patterns.php';


 
//Register Aione Slider
//require get_template_directory() . '/includes/register-aione-slider.php';

// Shortcodes
//require get_template_directory() . '/includes/shortcodes.php';

//Custom template tags for this theme.
require get_template_directory() . '/includes/template-tags.php';

//Functions which enhance the theme by hooking into WordPress.
require get_template_directory() . '/includes/template-functions.php';

//Customizer additions.
require get_template_directory() . '/includes/customizer.php';

//Load Jetpack compatibility file.
if ( defined( 'JETPACK__VERSION' ) ) {
	require get_template_directory() . '/includes/jetpack.php';
}

//Customize Menu
require get_template_directory() . '/includes/customize-menu.php';

// Aione custom function
require get_template_directory() . '/custom-functions.php';

// Aione Blocks
require get_template_directory() . '/blocks/main.php';


add_action( 'init', 'kses_unfiltered_html' );
function kses_unfiltered_html() {
    //$user = wp_get_current_user();

    if ( current_user_can('edit_pages') || current_user_can('edit_posts')){
        kses_remove_filters();
    }
}